platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -317,6 +332,38 @@ static OtaPalStatus_t otaPal_CheckFileSignature( OtaFileContext_t * const C )
     return OTA_PAL_COMBINE_ERR( mainErr, 0 );
 }
 
+static OtaPalPathGenStatus_t getFilePathFromCWD( char * pCompleteFilePath,
+                                                 const char * pFileName )
+{
+    char * pCurrentDir = NULL;
+    OtaPalPathGenStatus_t status = OtaPalFileGenSuccess;
+
+    /* Get current directory. */
+    pCurrentDir = getcwd( pCompleteFilePath, OTA_FILE_PATH_LENGTH_MAX - 1 );
+
+    if( pCurrentDir == NULL )
+    {
+        LogError( ( "Failed to get current working directory: %s", strerror( errno ) ) );
+        status = OtaPalCWDFailed;
+    }
+    else
+    {
+        /* Add the filename . */
+        if( strlen( pCompleteFilePath ) + strlen( pFileName ) + 2U > OTA_FILE_PATH_LENGTH_MAX )
+        {
+            LogError( ( "Insufficient space to generate file path" ) );
+            status = OtaPalBufferInsufficient;
+        }
+        else
+        {
+            strcat( pCompleteFilePath, "/" );
+            strncat( pCompleteFilePath, pFileName, strlen( pFileName ) );
```

Comment:
The lengths have already been checked above so I think it is more efficient to just do a simple strcat here. Strlen counts the entire length one char at a time a second time here for no additional benefit, we already know the string will fit be cause we checked above did we not?

Reply:
Thanks, addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1487/commits/8e4a008c671b3a39f450ac191c011630c4c01f51